Added documentation for profile tooling (#269)
microsoft/mu_feature_config: SetupDataPkg/Docs/Profiles/Overview.md

Alternatively, if the generic profile is chosen, ActiveProfileIndexSelectorLib will return MAX_UINT32 to indicdate the
gProfileData structure is not used for this boot and instead only the defaults in gKnobData (and possibly any
overrides found in variable storage) will be used.

### Platform Usage Recommendations

#### Generate Profile CSV Files

There are 2 ways to generate the CSV files:

- Through ConfigEditor UI tool:

  - Load the XML configuration file
  - Make any changes
  - Save the CSV file by selecting the "Save Config Changes to Change File" button. Show below:

    ![Profile Through ConfigEditor UI tool](Images/profile_ui.png)

- Through the command line:

  - Run the following command:

    ```bash
    GenNCCfgData  GENCSV  XmlFile[;BinFile]   CsvOutFile

>     ```

Reviewer: I think the command is simple enough, but you might also want to consider adding an actual example while you're here.

Author: Thanks, updated.

#### Add Profile CSV Files to PlatformBuild.py

If platforms would like to integrate the profile CSV files into their build, they can add the following to their build by
following the statements outlined in the [section above](#flow).

#### Provide Profile Names and IDs

Platforms also can provide a 2-character name and/or 1-byte ID for each profile. These are used to identify the profile
by the selector, which could be BMC, OS, or other.

> In firmware, the name and ID are provided in the `gProfileFlavorNames` and `gProfileFlavorIds`, respectively. Platforms

Reviewer 2: ```suggestion
In firmware, the name and ID are auto generated and can be accessed through `gProfileFlavorNames` and `gProfileFlavorIds`, respectively. Platforms

```

> can elect to create HOBs for these meta data information and provide them to the selector. See example [here](https://github.com/microsoft/mu_oem_sample/tree/release/202302/OemPkg/OemConfigPolicyCreatorPei).

Reviewer 2: ```suggestion
can elect to create HOBs for this meta data information and consume it in other drivers. See example [here](https://github.com/microsoft/mu_oem_sample/tree/release/202302/OemPkg/OemConfigPolicyCreatorPei).

```
